tools/cloudera-parcel-gcsconnector/create_parcel.sh

```diff
@@ -96,41 +208,81 @@ cat >>${filen^^}-$version/meta/parcel.json<< EOL
     "baseVersion":        "${filen^^}$version",
     "patchCount":         ""
   },
-
   "conflicts":          "",
-
   "setActiveSymlink":   true,
-
   "scripts": {
-        "defines": "${filen^^}.sh"
+    "defines": "${filen^^}.sh"
   },
-
   "packages": [ ],
   "components": [ ],
-
   "provides": [
-    "cdh-plugin"
+  "cdh-plugin"
   ],
-
   "users": { },
-
   "groups": [ ]
 }
-
 EOL
 
-##export HADOOP_CLASSPATH to enable command line commands can use the connector
-cat >>${filen^^}-$version/meta/$filen.sh<< EOL
+if [[ $? -ne 0 ]]; then
+  echo "Error: failed to write in parcel.json "$(pwd)/${filen^^}-$version/meta/parcel.json
+  graceful_exit
+fi
+
+##export HADOOP_CLASSPATH to enable command line to use the connector
+
+cat >>${filen^^}-$version/meta/$filen.sh<<EOL
+#!/bin/bash
 export HADOOP_CLASSPATH=\$HADOOP_CLASSPATH:/opt/cloudera/parcels/$filen-$version/lib/hadoop/lib/gcs-connector-latest-hadoop2.jar
 EOL
 
-## Create parcel file, checksum file and shift parcel files to cloudera parcel directory and change ownership to cloudera-scm user
-sudo tar zcvf ${filen^^}-$version-$OS.parcel ${filen^^}-$version/ --owner=root --group=root
+if [[ $? -ne 0 ]]; then
+  echo "Error: failed to write in file "$(pwd)/${filen^^}-$version/meta/$filen.sh
+  graceful_exit
+fi
+
+##Create parcel file, checksum file and move parcel files to
+##cloudera parcel directory and change ownership to cloudera-scm user
+
+if grep -q "cloudera-scm" /etc/group; then
```

Comment:
Can this be tested inside [[ ... ]]

https://google.github.io/styleguide/shell.xml?showone=Test,_%5B_and_%5B%5B#Test,_%5B_and_%5B%5B

Reply:
This one is a bash command. It does not really work with brackets single or double.

Reply:
What about...

grep -q "cloudera-scm" /etc/group
if [[ $? == 0 ]]; then
  ...
else
  ...
fi

Reply:
we tried $(getent group cloudera-scm). This works fine. Updating the code